src/embedding-utils.js

```diff
@@ -1,27 +1,58 @@
-import ReporterPluginHost from './reporter/plugin-host';
-import TestRunErrorFormattableAdapter from './errors/test-run/formattable-adapter';
-import * as testRunErrors from './errors/test-run';
-import TestRun from './test-run';
-import COMMAND_TYPE from './test-run/commands/type';
-import Assignable from './utils/assignable';
-import { getTestList, getTestListFromCode } from './compiler/test-file/formats/es-next/get-test-list';
-import { getTypeScriptTestList, getTypeScriptTestListFromCode } from './compiler/test-file/formats/typescript/get-test-list';
-import { getCoffeeScriptTestList, getCoffeeScriptTestListFromCode } from './compiler/test-file/formats/coffeescript/get-test-list';
-import { initSelector } from './test-run/commands/validations/initializers';
+const lazyRequire                    = require('import-lazy')(require);
+const hammerhead                     = lazyRequire('testcafe-hammerhead');
+const ReporterPluginHost             = lazyRequire('./reporter/plugin-host');
+const TestRunErrorFormattableAdapter = lazyRequire('./errors/test-run/formattable-adapter');
+const testRunErrors                  = lazyRequire('./errors/test-run');
+const TestRun                        = lazyRequire('./test-run');
```

Comment:
We should keep the regular `require` for `TestRun`. 
If I understand correctly, we have an issue with `import-lazy` here. 
When it calls Reflect.construct it requires the real TestRun constructor, so real  the `require` work occurs during object extending and we don't have a place where we can expand the prototype. 
So we miss `TestRun.prototype` which changes for the `RecordRun` inheritor.